// @better-auth/core/dist/instrumentation/api.mjs
//#region src/instrumentation/api.ts
function createNoopSpan() {
	return {
		end() {},
		setAttribute(_key, _value) {},
		setStatus(_status) {},
		recordException(_exception) {}
	};
}
function createNoopTracer(noopSpanInstance) {
	function startActiveSpan(_name, _options, fn) {
		return fn(noopSpanInstance);
	}
	return { startActiveSpan };
}
function createNoopTraceAPI() {
	const noopTracer = createNoopTracer(createNoopSpan());
	return { getTracer(_name, _version) {
		return noopTracer;
	} };
}
function createNoopOpenTelemetryAPI() {
	return {
		SpanStatusCode: {
			UNSET: 0,
			OK: 1,
			ERROR: 2
		},
		trace: createNoopTraceAPI()
	};
}
const noopOpenTelemetryAPI = createNoopOpenTelemetryAPI();
let openTelemetryAPIPromise;
let openTelemetryAPI;
function getOpenTelemetryAPI() {
	if (!openTelemetryAPIPromise) openTelemetryAPIPromise = import("@opentelemetry/api").then((mod) => {
		openTelemetryAPI = mod;
	}).catch(() => void 0);
	return openTelemetryAPI ?? noopOpenTelemetryAPI;
}
//#endregion
export { getOpenTelemetryAPI };
